src/python/WMCore/Agent/Daemon/Details.py

```diff
@@ -28,7 +31,7 @@ def run(command):
         stdin=subprocess.PIPE,
         )
 
-    proc.stdin.write(command)
+    proc.stdin.write(encodeUnicodeToBytesConditional(command, condition=PY3))
```

Comment:
This looks good to me! However, I will leave the full review to Todor, who already worked with xml and py3